

# This image belongs to a larger project called Bioinformatics Docker Images Project (http://pegi3s.github.io/dockerfiles)
## (Please note that the original software licenses still apply)

This image facilitates the usage of recombine, a docker image that can be used to infer recombination in a set of nucleotide sequences using the geneconv, bootscan, maxchi, siscan, chimaera, and rdp softwares, as implemented in [OpenRDP](https://github.com/PoonLab/OpenRDP), [3SEQ](https://mol.ax/software/3seq/) with the possibility of using custom ptables, and [Phipack](https://www.maths.otago.ac.nz/~dbryant/software/phimanual.pdf)(Pairwise Homoplasy Index (Phi), Maximum Ï‡2 (Max Chi^2) and the Neighbour Similarity Score (NSS)).

# Using the recombine image in Linux

In order to use the methods implemented in OpenRDP, a config file is needed. A standard config file can be generated with the command:

`docker run --rm -v /your/data/dir:/data pegi3s/recombine bash -c "/app/recombine config"`

In this command, you should replace:
- `/your/data/dir` to point to the working directory.

In order to use the 3SEQ approach, a ptable is needed. It can be generated with the following command:

`docker run --rm -v /your/data/dir:/data pegi3s/recombine bash -c "/app/recombine ptable size name"`

In this command, you should replace:
- `/your/data/dir` to point to the working directory.
- `size` to the actual size of the ptable to be generated. A size of 700 will generate a file that is approximately 440 Mb in size that can take up to half an hour to generate. Therefore, we advise the users to reuse the generated file.
- `name` to the actual name of the ptable to be generated.

To run all methods, you should adapt and run the following command:

`docker run --rm -v /your/data/dir:/data pegi3s/recombine bash -c "/app/recombine all file config ptable_name"`

In this command, you should replace:
- `/your/data/dir` to point to the directory that contains the input files you want to analyze.
- `file` to the name of the FASTA file containing the aligned nucleotide sequences.
- `config` to the name of the config file generated with the above command.
- `ptable_name` to the name of the ptable to be used by 3SEQ, generated by the above command.

All files must be in the working directory.

To run just one method, you should adapt and run the following command:

`docker run --rm -v /your/data/dir:/data pegi3s/recombine bash -c "/app/recombine method file config/ptable_name"`

In this command, you should replace:
- `/your/data/dir` to point to the directory that contains the input files you want to analyze.
- `method` to the name of the chosen method (the options are: geneconv, bootscan, maxchi, siscan, chimaera, rdp, 3seq, phipack).
- `file` to the name of the FASTA file containing the aligned nucleotide sequences.
- `config/ptable_name` to the name of the config (if using geneconv, bootscan, maxchi, siscan, chimaera, or rdp) or ptable (if using 3seq) file. If using phipack this parameter can be ommited.


